specifyweb/frontend/js_src/lib/components/DataModel/legacyTypes.ts

```diff
@@ -171,7 +171,10 @@ export type SpecifyResource<SCHEMA extends AnySchema> = {
     readonly errorOnAlreadySaving?: boolean;
   }): Promise<SpecifyResource<SCHEMA>>;
   destroy(): Promise<void>;
-  fetch(): Promise<SpecifyResource<SCHEMA>>;
+  fetch(options?: {
+    success?: (model: SpecifyResource<SCHEMA>, resoonse: unknown) => void;
+    error?: (model: SpecifyResource<SCHEMA>, response: unknown) => void;
+  }): Promise<SpecifyResource<SCHEMA>>;
```

Comment:
hmm, I don't see these being used in this PR
Also, I don't know how well these would work since these callbacks are handled by the backbone.
You might get by by using `hijackBackboneAjax` instead - see it's usages

Reply:
Yes, I was looking into possible solutions for this and came across those callbacks mentioned in the Backbone docs. 
I had figured it would be helpful to include them, even if they were not used anywhere (to save someone the trouble of looking through the Backbone docs).

However, I would prefer using hijackBackboneAjax to handle such cases anyways, to remain consistent and have more control!